sentry-ruby/lib/sentry/client.rb

```diff
@@ -183,6 +197,19 @@ def send_event(event, hint = nil)
       raise
     end
 
+    # Send an envelope directly to Sentry.
+    # @param envelope [Envelope] the envelope to be sent.
+    # @return [void]
+    def send_envelope(envelope)
+      transport.send_envelope(envelope) if configuration.sending_to_dsn_allowed?
+      spotlight_transport.send_envelope(envelope) if spotlight_transport
+    rescue => e
+      # note that we don't record client reports for direct envelope types
+      # such as metrics, sessions etc
```

Comment:
We do have [data categories](https://develop.sentry.dev/sdk/rate-limiting/#definitions) for both sessions and metrics, though.